pylint/checkers/variables.py

```diff
@@ -844,7 +844,13 @@ def visit_global(self, node):
 
         module = frame.root()
         default_message = True
+        locals_ = node.scope().locals
         for name in node.names:
+            if any(isinstance(local, astroid.node_classes.Import)
```

Comment:
I think we should move this to the line 861, right before the `not assign_nodes` checks because otherwise it is possible for us to not emit `global-variable-undefined` for instance